File: 프로그래머스/unrated/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public static int lcm(int a,int b){
        int min = Math.min(a,b);
        int mul = 1;
        for(int i=1;i<=min;i++){
            if(a%i==0 && b%i==0){
                mul= mul*i;
                a/=i;
                b/=i;
            }
        }      
        return a*b*mul;
    }
    public static int gcd(int a, int b){
        int min = Math.min(a,b);
        int i;
        int result=0;
        for(i=min;i>0;i--){
            if(a%i==0 && b%i==0){
                result = i;
                break;
            }
        }
        return result;
    }
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int[] answer = new int[2];
        int denom = denom1*denom2;
        int numer = numer1*denom2+numer2*denom1;
        answer[0]=numer/gcd(numer,denom);
        answer[1]=denom/gcd(numer,denom);
        return answer;
    }
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.cs
using System;

public class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        // 분모는 최소공배수
        int lcm = calcLCM(denom1, denom2);
        
        // 분자 계산
        int numer1Adjusted = numer1 * (lcm / denom1);
        int numer2Adjusted = numer2 * (lcm / denom2);
        int resultNumer = numer1Adjusted + numer2Adjusted;
        
        // 기약분수로 만들기
        int gcdResult = calcGCD(resultNumer, lcm);
        int[] answer = new int[] { resultNumer / gcdResult, lcm / gcdResult };
        
        return answer;
    }
    
    public int calcLCM(int a, int b) {
        return (a * b) / calcGCD(a, b);
    }
    
    public int calcGCD(int a, int b) {
        while (b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
}